lib/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,268 @@
+import React, { Component, ReactNode, AllHTMLAttributes } from 'react';
+import classnames from 'classnames';
+import { Omit } from 'utility-types';
+import ThemeConsumer from '../ThemeConsumer/ThemeConsumer';
+import getCheckboxRadioSize from '../private/getCheckboxRadioSize';
+import Box from '../Box/Box';
+import Text from '../Text/Text';
+import FieldMessage from '../FieldMessage/FieldMessage';
+import TickIcon from '../icons/TickIcon/TickIcon';
+import styles from './Checkbox.css.js';
+import { px } from '../../themes/utils/toUnit';
+
+const textColorForState = ({
+  disabled,
+  hovered
+}: {
+  disabled: boolean | undefined;
+  hovered: boolean | undefined;
+}) => {
+  if (disabled) {
+    return 'secondary';
+  }
+
+  if (hovered) {
+    return 'formAccent';
+  }
+
+  return 'neutral';
+};
+
+export interface CheckboxProps
+  extends Omit<AllHTMLAttributes<HTMLElement>, 'label'> {
+  variant?: 'default' | 'inChecklistCard';
+  id: string;
+  checked: boolean;
+  label: ReactNode;
+  disabled?: boolean;
+  inputProps?: object;
+  labelProps?: object;
+  tone?: 'neutral' | 'critical' | 'positive';
+  message?: ReactNode | false;
+  messageProps?: object;
+  children?: ReactNode;
+}
+
+interface State {
+  hovered: boolean;
+}
+
+export default class Checkbox extends Component<CheckboxProps, State> {
+  static displayName = 'Checkbox';
+
+  constructor(props: CheckboxProps) {
+    super(props);
+
+    this.state = {
+      hovered: false
+    };
+
+    this.handleMouseOver = this.handleMouseOver.bind(this);
+    this.handleMouseOut = this.handleMouseOut.bind(this);
+  }
+
+  handleMouseOver() {
+    this.setState({ hovered: true });
+  }
+
+  handleMouseOut() {
+    this.setState({ hovered: false });
+  }
+
+  render() {
+    return (
+      <ThemeConsumer>
+        {theme => {
+          const {
+            variant,
+            id,
+            checked,
+            label,
+            disabled,
```

Comment:
Thoughts on making `disabled` have a default value of `false`? I think that'd be cleaner.